### DISCUSSION TITLE
Binary files firmware.bin and opendtu-onbattery-generic.bin differ

### DISCUSSION THREAD

**Discussion by IngoEF** [hoylabs/OpenDTU-OnBattery#197] [General]
Building an own binary in VS creates a firmware.bin which stucks between 31 and 33% when trying to flash it via OTA.
Erasing fully and flash the binaries via platformIO afterwards provides a working system. Why is the firmware.bin not flashable via OTA?
Am I missing something?

**Comment by madmartin** [comment]
Hello @IngoEF 
I remember I had the same effect (stuck upload) some weeks ago and do not exactly remember how I fixed other than flashing through USB. It did not occur again since then.
Self-compiled binaries are definitely flashable by OTA, there is no difference. When you download pre-compiled firmware from github, you must unpack the *zip, and select the largest firmware.bin for OTA.

**Comment by helgeerbe** [comment]
I observed this behavior sometimes. A reboot before OTA update can help. I can't say what the root cause is, but sometimes openDTU seems to be short on large memory blocks (memory fragmentation).
